modules/generic/src/main/scala/pureconfig/generic/EnumerationConfigReaderBuilder.scala

```diff
@@ -13,15 +15,19 @@ import shapeless.labelled._
  * @tparam A the type of objects capable of being read as an enumeration
  */
 trait EnumerationConfigReaderBuilder[A] {
```

Comment:
Like other auxiliary type classes for generic derivation, we should make this trait `private[generic]` so that we can more confidently change its API.

Reply:
I don't think we can make this `private[generic]` because it's part of the interface of the [public `deriveEnumerationReader` method](https://github.com/pureconfig/pureconfig/blob/3bb840f0d8d3a1c2d028b3d7215271d672f5e419/modules/generic/src/main/scala/pureconfig/generic/semiauto.scala#L23). When I do make it `private[generic]`, compiling the main code succeeds (😱 🙀 😱 ) but semiauto derivation fails for `Color` in `EnumerationSuites`.

Am I missing a trick here?